openverse_catalog/dags/providers/provider_api_scripts/nypl.py

```diff
@@ -1,211 +1,248 @@
+        creator = self._get_creators(name_properties) if name_properties else None
 
-def _get_capture_details(captures=None, metadata=None, creator=None, title=None):
-    if captures is None:
-        captures = []
-    for img in captures:
-        image_id = img.get("imageID", {}).get("$")
-        if image_id is None:
-            continue
-        image_url = _get_image_url(img.get("imageLinks", {}).get("imageLink", []))
-        foreign_landing_url = img.get("itemLink", {}).get("$")
-        license_url = img.get("rightsStatementURI", {}).get("$")
-        if image_url is None or foreign_landing_url is None or license_url is None:
-            continue
-
-        image_store.add_item(
-            foreign_identifier=image_id,
-            foreign_landing_url=foreign_landing_url,
-            image_url=image_url,
-            license_info=get_license_info(license_url=license_url),
-            title=title,
-            creator=creator,
-            meta_data=metadata,
+        metadata = self._get_metadata(mods)
+        category = (
+            ImageCategory.PHOTOGRAPH.value
+            if metadata.get("genre") == "Photographs"
+            else None
         )
 
-
-def _get_title(titleinfo):
-    title = None
-    if type(titleinfo) == list and len(titleinfo) > 0:
-        title = titleinfo[0].get("title", {}).get("$")
-    return title
-
-
-def _get_creators(creatorinfo):
-    if type(creatorinfo) == list:
+        captures = item_details.get("sibling_captures", {}).get("capture")
+        if not captures:
+            return None
+        if not isinstance(captures, list):
+            captures = [captures]
+        images = []
+        for capture in captures:
+            image_id = capture.get("imageID", {}).get("$")
+            if image_id is None:
+                continue
+
+            image_link = capture.get("imageLinks", {}).get("imageLink", [])
+            image_url, filetype = self._get_image_data(image_link)
+            if not image_url:
+                continue
+
+            foreign_landing_url = capture.get("itemLink", {}).get("$")
+            license_url = capture.get("rightsStatementURI", {}).get("$")
+            if not foreign_landing_url or license_url is None:
+                continue
+
+            image_data = {
+                "foreign_identifier": image_id,
+                "foreign_landing_url": foreign_landing_url,
+                "image_url": image_url,
+                "license_info": get_license_info(license_url=license_url),
+                "title": title,
+                "creator": creator,
+                "filetype": filetype,
+                "category": category,
+                "meta_data": metadata,
+            }
+            images.append(image_data)
+        return images or None
+
+    @staticmethod
+    def _get_filetype(description: str):
+        """
+        Extracts the filetype from a description string like:
+        "Cropped .jpeg (1600 pixels on the long side)"
```

Comment:
Maybe we could add this or something similar?
```suggestion
        "Cropped .jpeg (1600 pixels on the long side)"
        This is required because the filetype is not present/extractable from the
        url via the MediaStore class.
```